src/kfs_grpc_inference_service.cpp

```diff
@@ -267,38 +261,27 @@ Status KFSInferenceServiceImpl::ModelInferImpl(::grpc::ServerContext* context, c
     }
     if (!status.ok()) {
         if (modelInstance) {
-            INCREMENT_IF_ENABLED(modelInstance->getMetricReporter().requestSuccessGrpcModelInfer);
+            INCREMENT_IF_ENABLED(modelInstance->getMetricReporter().requestFailGrpcModelInfer);
         }
         SPDLOG_DEBUG("Getting modelInstance or pipeline failed. {}", status.string());
         return status;
     }
 
     if (pipelinePtr) {
-        status = pipelinePtr->execute(ExecutionContext(
-            ExecutionContext::Interface::GRPC,
-            ExecutionContext::Method::ModelInfer));
-        if (status.ok()) {
-            INCREMENT_IF_ENABLED(pipelinePtr->getMetricReporter().requestSuccessGrpcModelInfer);
-        } else {
-            INCREMENT_IF_ENABLED(pipelinePtr->getMetricReporter().requestFailGrpcModelInfer);
-        }
+        reporterOut = &pipelinePtr->getMetricReporter();
+        status = pipelinePtr->execute(executionContext);
     } else {
+        reporterOut = &modelInstance->getMetricReporter();
         status = modelInstance->infer(request, response, modelInstanceUnloadGuard);
-        if (status.ok()) {
-            INCREMENT_IF_ENABLED(modelInstance->getMetricReporter().requestSuccessGrpcModelInfer);
-        } else {
-            INCREMENT_IF_ENABLED(modelInstance->getMetricReporter().requestFailGrpcModelInfer);
-        }
     }
 
+    INCREMENT_IF_ENABLED(reporterOut->getInferRequestMetric(executionContext, status.ok()));
```

Comment:
What is this metric? What information it gives?

Reply:
requests_success and requests_fail - depending on `status.ok()` outcome